server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -177,6 +177,9 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
 
     protected final ThreadPool threadPool;
 
+    public static final String STATELESS_SHARD_THREAD_NAME = "stateless_shard";
+    public static final String STATELESS_TRANSLOG_THREAD_NAME = "stateless_translog";
```

Comment:
Does it really need to be public?

Reply:
I think so, we want to refer to it from the stateless plugin. At least, I'd prefer to do it this way rather than to duplicate the string literal.

Reply:
Oh right